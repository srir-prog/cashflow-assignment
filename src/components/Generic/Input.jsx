import React from "react";
const Input = ({ value, onChange, type, placeholder, className, name }) => {
  return (
    <input
      name={name}
      type={type}
      className={className}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
    />
  );
};

export default Input;
