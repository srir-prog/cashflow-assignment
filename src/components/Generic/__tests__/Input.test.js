import React from 'react';
import { shallow } from 'enzyme';
import Input from './../Input';
import renderer from 'react-test-renderer';

describe('Input component', () => {
    let component;
    const props = {
        name: 'input',
        type: 'text',
        className: 'className',
        placeholder: 'placeholder',
        value: 'Income',
        onChange: jest.fn()
    }
    beforeEach(() => {
        component = shallow(<Input {...props} />);
    });
    it('should render correctly', () => {
        //Snapshot testing.
        const tree = renderer.create(<Input {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('should render <input> tag', () => {
        const inputElement = component.find('input');
        expect(inputElement.length).toBe(1);
    });
});