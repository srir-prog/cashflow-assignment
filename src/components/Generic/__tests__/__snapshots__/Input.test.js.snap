// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<input
  className="className"
  name="input"
  onChange={[MockFunction]}
  placeholder="placeholder"
  type="text"
  value="Income"
/>
`;
